daemon/execdriver/native/seccomp_default.go

```diff
@@ -0,0 +1,332 @@
+// +build linux
+
+package native
+
+import "github.com/opencontainers/runc/libcontainer/configs"
+
+var defaultSeccompProfile = &configs.Seccomp{
+	DefaultAction: configs.Allow,
+	Syscalls: []*configs.Syscall{
+		{
+			// Quota and Accounting syscalls which could let containers
+			// disable their own resource limits or process accounting
+			Name:   "acct",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Prevent containers from using the kernel keyring,
+			// which is not namespaced
+			Name:   "add_key",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Similar to clock_settime and settimeofday
+			// Time/Date is not namespaced
+			Name:   "adjtimex",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Time/Date is not namespaced
+			Name:   "clock_settime",
+			Action: configs.Errno,
+			Args:   []*configs.Arg{},
+		},
+		{
+			// Deny cloning new namespaces
+			Name:   "clone",
+			Action: configs.Errno,
+			Args: []*configs.Arg{
+				{
+					// flags from sched.h
+					// CLONE_NEWUTS		0x04000000
+					// CLONE_NEWIPC		0x08000000
+					// CLONE_NEWUSER	0x10000000
+					// CLONE_NEWPID		0x20000000
+					// CLONE_NEWNET		0x40000000
+					Index: 0,
+					Value: uint64(0x04000000),
+					Op:    configs.GreaterThanOrEqualTo,
+				},
+			},
+		},
+		{
+			// Deny cloning new namespaces
+			Name:   "clone",
```

Comment:
oh ya sorry meant to clean that up

On Tue, Dec 22, 2015 at 10:04 AM, Matthew Heon <notifications@github.com>
wrote:

> In daemon/execdriver/native/seccomp_default.go
> <https://github.com/docker/docker/pull/18780#discussion_r48279824>:
>
> > +				{
> > +					// flags from sched.h
> > +					// CLONE_NEWUTS		0x04000000
> > +					// CLONE_NEWIPC		0x08000000
> > +					// CLONE_NEWUSER	0x10000000
> > +					// CLONE_NEWPID		0x20000000
> > +					// CLONE_NEWNET		0x40000000
> > +					Index: 0,
> > +					Value: uint64(0x04000000),
> > +					Op:    configs.GreaterThanOrEqualTo,
> > +				},
> > +			},
> > +		},
> > +		{
> > +			// Deny cloning new namespaces
> > +			Name:   "clone",
>
> It looks like this can be combined with the previous Clone entry - just
> add the argument here to the array in the first one
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/18780/files#r48279824>.
>